src/Aspire.Hosting.NodeJs/NodeExtensions.cs

```diff
@@ -207,50 +196,147 @@ public static IResourceBuilder<ViteAppResource> AddViteApp(this IDistributedAppl
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<TResource> WithPnpm<TResource>(this IResourceBuilder<TResource> resource, bool install = false) where TResource : NodeAppResource
+    {
+        resource.WithCommand("pnpm")
+            .WithAnnotation(new JavaScriptInstallCommandAnnotation("pnpm", ["install"]))
+            .WithAnnotation(new JavaScriptRunCommandAnnotation(["run"]))
+            .WithAnnotation(new JavaScriptBuildCommandAnnotation("pnpm", ["run", "build"]));
+
+        AddInstaller(resource, install);
+        return resource;
+    }
+
+    /// <summary>
+    /// Configures the Node.js resource to run the command to install packages before the application starts.
+    /// </summary>
+    /// <param name="resource">The NodeAppResource.</param>
+    /// <param name="command">The executable command name</param>
+    /// <param name="args">The command line arguments for the JavaScript package manager's install command.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<TResource> WithInstallCommand<TResource>(this IResourceBuilder<TResource> resource, string command, string[] args) where TResource : NodeAppResource
+    {
+        resource.WithAnnotation(new JavaScriptInstallCommandAnnotation(command, args));
+
+        AddInstaller(resource, install: true);
+        return resource;
+    }
 
-        // Only install packages during development, not in publish mode
+    /// <summary>
+    /// Configures the Node.js resource to run the command to build the app during deployment.
+    /// </summary>
+    /// <param name="resource">The NodeAppResource.</param>
+    /// <param name="command">The executable command name</param>
+    /// <param name="args">The command line arguments for the JavaScript package manager's build command.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<TResource> WithBuildCommand<TResource>(this IResourceBuilder<TResource> resource, string command, string[] args) where TResource : NodeAppResource
+    {
+        return resource.WithAnnotation(new JavaScriptBuildCommandAnnotation(command, args));
+    }
+
+    private static void AddInstaller<TResource>(IResourceBuilder<TResource> resource, bool install) where TResource : NodeAppResource
+    {
+        // Only install packages if not in publish mode
         if (!resource.ApplicationBuilder.ExecutionContext.IsPublishMode)
         {
-            var installerName = $"{resource.Resource.Name}-npm-install";
-            var installer = new NodeInstallerResource(installerName, resource.Resource.WorkingDirectory);
+            // Check if the installer resource already exists
+            var installerName = $"{resource.Resource.Name}-installer";
```

Comment:
We have a really nice helper for this now https://github.com/dotnet/aspire/blob/d9c326677bb6d274e5bb9816a432c32a3dc25eb1/src/Aspire.Hosting/DistributedApplicationBuilderExtensions.cs#L91